torch/_inductor/cudagraph_trees.py

```diff
@@ -1569,7 +1576,7 @@ def _allocate_and_copy_recording_inputs(
 
         return recording_inputs
 
-    def check_invariants(self, inputs: List[Tensor]) -> bool:
+    def check_invariants(self, inputs: List[Tensor]) -> CheckInvariantStatus:
```

Comment:
Could we return both the `CheckInvariantStatus` and also a function which would generate a more detailed message (to be used lazily) ?

Would be great to add additional info with stack traces of which inputs changed